docs/scripts/util/checkLink.ts

```diff
@@ -0,0 +1,81 @@
+import https from 'https';
+import http from 'http';
+import { IGNORED_LINKS } from '../../src/data/ignoredLinks';
+
+const requestedUrl: Set<string> = new Set();
+export async function checkLink(
+  {
+    href,
+    tagName,
+    tagText,
+    pageIdx,
+    pageUrl,
+  }: {
+    href: string;
+    tagName: string;
+    tagText: string;
+    pageIdx: string;
+    pageUrl: string;
+  },
+  linkIdx: number
+) {
+  return new Promise(async (res, rej) => {
+    if (!href) {
+      console.log(
+        `⚠️[WARNING...] page #${pageIdx} link #${linkIdx} "${tagName}" tag "${tagText}" doesn't have a href.`
+      );
+      res(0);
+    } else if (IGNORED_LINKS.includes(href) || requestedUrl.has(href)) {
+      console.log(
+        `⏭[SKIPPING...] page #${pageIdx} link #${linkIdx} ${href} from ${tagName} tag "${tagText}" on page ${pageUrl}, because it is on the IGNORED_LINKS list or have already been requested.`
+      );
+      res(0);
+    } else {
+      const { get } = href.includes('https:') ? https : http;
+      const request = await get(href, async ({ statusCode = 0 }) => {
+        await returnStatus({ statusCode, href });
+        requestedUrl.add(href);
+        res(statusCode);
+      });
+      request.end();
+    }
+  });
+
+  async function returnStatus({
+    statusCode,
+    href,
+  }: {
+    statusCode: number;
+    href: string;
+  }) {
+    if ([200, 301, 303, 308].includes(statusCode)) {
+      console.log(
+        `↩️ [RETURNING STATUS...] ${statusCode} page #${pageIdx} link #${linkIdx} -- ${href} from ${tagName} tag "${tagText}" on page ${pageUrl}`
+      );
+
+      /**
+       * If 308, check if it's a internal direction (see docs/next.config.js redirects logic)
+       * If it's internal direction, after adding the platform, it should be 200
+       * Otherwise, the link needs to be updated
+       */
+      if (statusCode === 308) {
+        const hostNameRegex = /http(s)?:\/\/[^/]*/i;
+        const platform = pageUrl.replace(hostNameRegex, '').split('/')[1];
```

Comment:
why not just 
```suggestion
        const platform = pageUrl.replace('https://', '').split('/')[1];
```

Reply:
because it matches the entire hostname, e.g., "http://github.com/"

Reply:
add a comment e7ecf9b5b9ae41828a4d0d4d9381977ab2ae574a